src/raft_server.c

```diff
@@ -6223,7 +6240,12 @@ raft_server_instance_run(const char *raft_uuid_str,
     if (!raft_uuid_str || !this_peer_uuid_str || !sm_request_handler)
         return -EINVAL;
 
-    struct raft_instance *ri = NULL;
+    struct raft_instance *ri = raft_net_init_instance();
+    if (!ri)
+            return -ENOENT;
+
+    raft_server_global_instance_init(ri);
```

Comment:
There should be a destructor method which corresponds to `raft_server_global_instance_init()` and it should call:
`raft_net_destroy_instance(ri)`.